src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -0,0 +1,354 @@
+                            "buffer": element["digital_inputs"]["buffer"],
+                        }
+                    }
+                if "digital_outputs" in element:
+                    element_dict["digitalOutputs"] = {
+                        "port": (element["digital_outputs"]["controller"], element["digital_outputs"]["port"]),
+                    }
+
+                elements[bus_name] = element_dict
+
+            return elements, mixers
+
+    settings: QuantumMachinesClusterSettings
+    device: QMMDriver
+    qmm: QuantumMachinesManager
+    qm: QuantumMachine
+    config: DictQuaConfig
+    octave_config: QmOctaveConfig | None = None
+
+    def _is_connected_to_qm(self):
+        """Check if a connection to Quantum Machine is open."""
+        return hasattr(self, "qm")
+
+    @Instrument.CheckDeviceInitialized
+    def initial_setup(self):
+        """Sets initial instrument settings.
+
+        Creates an instance of the Qilililab Quantum Machines Manager, and sets the configuration dictionary.
+        """
+        if self.settings.octaves:
+            self.octave_config = QmOctaveConfig()
+            self.octave_config.set_calibration_db(os.getcwd())
+            for octave in self.settings.octaves:
+                self.octave_config.add_device_info(octave["name"], self.settings.address, octave["port"])
+        self.qmm = QuantumMachinesManager(
+            host=self.settings.address, cluster_name=self.settings.cluster, octave=self.octave_config
+        )
+
+        self.config = self.settings.to_qua_config()
+
+    @Instrument.CheckDeviceInitialized
+    def turn_on(self):
+        """Turns on the instrument."""
+        if not self._is_connected_to_qm():
+            self.qm = self.qmm.open_qm(config=self.config, close_other_machines=True)
+
+            if self.settings.run_octave_calibration:
+                self.run_octave_calibration()
+
+    @Instrument.CheckDeviceInitialized
+    def reset(self):
+        """Resets instrument settings."""
+
+    @Instrument.CheckDeviceInitialized
+    def turn_off(self):
+        """Turns off an instrument."""
+        if self._is_connected_to_qm():
+            self.qm.close()
+
+    def update_configuration(self, compilation_config: dict):
+        """Update the configuration dictionary by appending the configuration generated by compilation."""
+        self.config = merge_dictionaries(dict(self.config), compilation_config)
+        # If we are already connected, reopen the connection with the new configuration
+        if self._is_connected_to_qm():
+            self.qm = self.qmm.open_qm(config=self.config, close_other_machines=True)
+
+    def run_octave_calibration(self):
+        """Run calibration procedure for the buses with octaves, if any."""
+        elements = [element for element in self.config["elements"] if "RF_inputs" in self.config["elements"][element]]
+        for element in elements:
+            self.qm.calibrate_element(element)
```

Comment:
In here it does complain that the instance of QuantumMachine does not have the calibrate_element attribute. However this calibration was worked in the lab, wasn't it?

Reply:
Same here, you have to update to `1.1.6`.

Reply:
Done